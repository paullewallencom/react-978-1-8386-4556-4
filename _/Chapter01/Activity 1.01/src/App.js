import React, { Component } from "react";

const Header = props => <h1>{props.title}</h1>;

const InventoryItem = props => (
  <div className="InventoryItem">
    <strong>{props.itemName}</strong>
    <hr />
    <p>{props.itemPrice}</p>
  </div>
);

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [
        { itemName: "Shoe", itemPrice: 5 },
        { itemName: "Sock", itemPrice: 3 }
      ]
    };
  }
  render() {
    return (
      <div className="App">
        <Header title="BuyStuff" />
        <InventoryItem
          itemName={this.state.items[0].itemName}
          itemPrice={this.state.items[0].itemPrice}
        />
        <InventoryItem
          itemName={this.state.items[1].itemName}
          itemPrice={this.state.items[1].itemPrice}
        />
      </div>
    );
  }
}

export default App;
