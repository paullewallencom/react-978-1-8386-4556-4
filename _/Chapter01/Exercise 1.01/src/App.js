import React from "react";

class App extends React.Component {
  render() {
    return (
      <div className="Example" id="my-element">
        <div
          className="greeting"
          style={{ background: "black", color: "white" }}
          onClick={() => alert("hello")}
        >
          Hello World
        </div>
      </div>
    );
  }
}

export default App;
