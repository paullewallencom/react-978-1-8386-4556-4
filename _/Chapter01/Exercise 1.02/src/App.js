import React from "react";

const App = props => (
  <div>
    <h1>My App</h1>
    <br />
    Hello React!
  </div>
);

export default App;
