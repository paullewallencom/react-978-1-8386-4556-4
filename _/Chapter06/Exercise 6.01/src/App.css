@import url("https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css");

.dark-theme {
  background: #333;
  color: #fff;
}
.dark-theme .jumbotron {
  background: #444;
}
