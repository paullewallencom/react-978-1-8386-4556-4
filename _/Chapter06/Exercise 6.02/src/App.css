.container {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 10%;
}
button {
  margin-top: 2rem;
  border: solid 1px #999;
  padding: 0.5rem 1rem;
  cursor: pointer;
}
