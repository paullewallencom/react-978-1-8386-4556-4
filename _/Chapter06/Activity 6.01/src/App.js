import React, { Component } from "react";
// styles
import "./App.css";
// data
import productData from "./products.json";
// components
import { ProductListing } from "./components/ProductListing";
import { Tags } from "./components/Tags";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedTag: null
    };
    this.handleTags = this.handleTags.bind(this);
  }
  handleTags(tagName) {
    return () => {
      this.setState({ selectedTag: tagName });
    };
  }
  render() {
    const { products, ingredients } = productData;
    const { selectedTag } = this.state;

    return (
      <div className="container">
        <h1>Products</h1>
        <Tags
          tags={ingredients}
          selectedTag={selectedTag}
          handleTags={this.handleTags}
        />
        <ProductListing products={products} selectedTag={selectedTag} />
      </div>
    );
  }
}

export default App;
