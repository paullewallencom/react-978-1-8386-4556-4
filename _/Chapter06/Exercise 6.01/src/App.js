import React, { Component } from "react";
// styles
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      theme: "light"
    };
    this.toggleTheme = this.toggleTheme.bind(this);
  }
  toggleTheme() {
    const theme = this.state.theme === "light" ? "dark" : "light";
    this.setState({ theme });
  }
  render() {
    return (
      <div className={`${this.state.theme}-theme`}>
        <div className="jumbotron">
          <div className="container">
            <h1>Hello, world!</h1>
            <p>
              This is a template for a simple marketing or informational
              website. It includes a large callout called a jumbotron and three
              supporting pieces of content. Use it as a starting point to create
              something more unique.
            </p>
            <p>
              <button
                className="btn btn-primary btn-lg"
                onClick={this.toggleTheme}
              >
                Switch Theme
              </button>
            </p>
          </div>
        </div>

        <div className="container">
          <div className="row">
            <div className="col-md-8">
              <h2>State example</h2>
              <p className="lead">
                Adipisci architecto blanditiis cumque dolores doloribus dolorum
                eligendi ex nesciunt quisquam. Assumenda, dolorem enim eos est
                illum ipsa odit rem tenetur. Blanditiis!
              </p>
              <p>
                Adipisci aut dolore doloribus dolorum eius enim exercitationem
                facere iusto, nesciunt nostrum porro, quidem suscipit velit.
                Beatae blanditiis cum id quaerat quisquam.
              </p>
              <p>
                Accusantium ad, animi blanditiis dignissimos eius enim est fugit
                minima molestiae necessitatibus nostrum odio quae quaerat quidem
                quos rerum sint ullam voluptatibus.
              </p>
              <p>
                Beatae eius numquam pariatur ratione sapiente velit! Accusantium
                adipisci aut est hic illum, maxime molestias odit optio
                reiciendis sint tenetur, ut veritatis!
              </p>
              <ul>
                <li>
                  Aliquam amet eum exercitationem facere illo, illum ipsam
                  laboriosam modi nihil obcaecati odit optio provident, quae
                  quas quia quibusdam ratione totam veniam!
                </li>
                <li>
                  Asperiores, debitis eius harum, id incidunt natus quisquam
                  quos repellat tempore ut veniam voluptas. Adipisci aliquam
                  deleniti ea iusto quasi totam voluptatum!
                </li>
                <li>
                  Ad cupiditate, eos hic obcaecati voluptas voluptatibus!
                  Accusamus eaque laudantium modi quae repellat reprehenderit,
                  vel! Culpa doloremque eligendi excepturi minima nam sed!
                </li>
              </ul>
            </div>
            <div className="col-md-4">
              <todo-component></todo-component>
              <h2>Heading</h2>
              <p>
                Donec id elit non mi porta gravida at eget metus. Fusce dapibus,
                tellus ac cursus commodo, tortor mauris condimentum nibh, ut
                fermentum massa justo sit amet risus. Etiam porta sem malesuada
                magna mollis euismod. Donec sed odio dui.{" "}
              </p>
              <hr />
              <h2>Heading</h2>
              <p>
                Donec id elit non mi porta gravida at eget metus. Fusce dapibus,
                tellus ac cursus commodo, tortor mauris condimentum nibh, ut
                fermentum massa justo sit amet risus. Etiam porta sem malesuada
                magna mollis euismod. Donec sed odio dui.{" "}
              </p>
              <hr />
              <h2>Heading</h2>
              <p>
                Donec sed odio dui. Cras justo odio, dapibus ac facilisis in,
                egestas eget quam. Vestibulum id ligula porta felis euismod
                semper. Fusce dapibus, tellus ac cursus commodo, tortor mauris
                condimentum nibh, ut fermentum massa justo sit amet risus.
              </p>
            </div>
          </div>
          <hr />
          <footer>
            <p>&copy; 2016 Company, Inc.</p>
          </footer>
        </div>
      </div>
    );
  }
}

export default App;
