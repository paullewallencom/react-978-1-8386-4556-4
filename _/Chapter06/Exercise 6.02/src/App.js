import React, { Component } from "react";
// styles
import "./App.css";
// component
import { Modal } from "./components/Modal";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false
    };
    this.toggleModal = this.toggleModal.bind(this);
  }
  toggleModal() {
    this.setState({
      showModal: !this.state.showModal
    });
  }
  render() {
    return (
      <div className="container">
        <div>
          <p>
            Crema caffeine, single origin variety coffee a qui turkish. Wings
            strong siphon extra percolator sweet single origin percolator.
            Redeye aftertaste strong single shot organic redeye brewed
            kopi-luwak caffeine variety. Aftertaste, arabica crema to go
            trifecta that grinder.
          </p>
          <p>
            Flavour saucer, con panna sit wings macchiato dripper sugar qui
            lungo turkish dripper. Acerbic blue mountain milk ristretto crema,
            rich barista ut and cappuccino filter. Saucer dark cappuccino,
            froth, extraction coffee crema white half and half single shot to
            go. Flavour cup robust cream eu café au lait extraction con panna
            trifecta body blue mountain decaffeinated.
          </p>
          <p>
            Eu shop, mocha body cortado pumpkin spice galão. Java affogato
            ristretto, dark flavour qui est white grounds a black. Cultivar
            siphon flavour, extra organic grinder kopi-luwak acerbic java
            organic. To go, et trifecta irish percolator cream est ut single
            origin mocha.
          </p>
          <p>
            Qui beans, caffeine skinny bar seasonal mazagran. Turkish spoon
            beans rich arabica medium kopi-luwak turkish. A, con panna blue
            mountain aftertaste steamed, so irish crema caffeine rich.
            Aftertaste eu aftertaste foam, con panna and rich robust and aroma
            arabica.
          </p>
        </div>
        <button onClick={this.toggleModal}>Show Modal</button>
        <Modal showModal={this.state.showModal} toggleModal={this.toggleModal}>
          Text in overlay
        </Modal>
      </div>
    );
  }
}

export default App;
