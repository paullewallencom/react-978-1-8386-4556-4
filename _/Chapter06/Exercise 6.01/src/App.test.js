import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders app with light theme", () => {
  const { container } = render(<App />);
  expect(container.firstChild.classList.contains("light-theme")).toBe(true);
});

it("switches to dark theme on single click", () => {
  const { container, getByText } = render(<App />);
  fireEvent.click(getByText("Switch Theme"));
  expect(container.firstChild.classList.contains("dark-theme")).toBe(true);
});

it("switches to light theme on two clicks theme", () => {
  const { container, getByText } = render(<App />);
  fireEvent.click(getByText("Switch Theme"));
  fireEvent.click(getByText("Switch Theme"));
  expect(container.firstChild.classList.contains("light-theme")).toBe(true);
});
