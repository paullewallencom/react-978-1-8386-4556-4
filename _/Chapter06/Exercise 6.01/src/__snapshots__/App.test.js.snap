// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="light-theme"
  >
    <div
      class="jumbotron"
    >
      <div
        class="container"
      >
        <h1>
          Hello, world!
        </h1>
        <p>
          This is a template for a simple marketing or informational website. It includes a large callout called a jumbotron and three supporting pieces of content. Use it as a starting point to create something more unique.
        </p>
        <p>
          <button
            class="btn btn-primary btn-lg"
          >
            Switch Theme
          </button>
        </p>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-8"
        >
          <h2>
            State example
          </h2>
          <p
            class="lead"
          >
            Adipisci architecto blanditiis cumque dolores doloribus dolorum eligendi ex nesciunt quisquam. Assumenda, dolorem enim eos est illum ipsa odit rem tenetur. Blanditiis!
          </p>
          <p>
            Adipisci aut dolore doloribus dolorum eius enim exercitationem facere iusto, nesciunt nostrum porro, quidem suscipit velit. Beatae blanditiis cum id quaerat quisquam.
          </p>
          <p>
            Accusantium ad, animi blanditiis dignissimos eius enim est fugit minima molestiae necessitatibus nostrum odio quae quaerat quidem quos rerum sint ullam voluptatibus.
          </p>
          <p>
            Beatae eius numquam pariatur ratione sapiente velit! Accusantium adipisci aut est hic illum, maxime molestias odit optio reiciendis sint tenetur, ut veritatis!
          </p>
          <ul>
            <li>
              Aliquam amet eum exercitationem facere illo, illum ipsam laboriosam modi nihil obcaecati odit optio provident, quae quas quia quibusdam ratione totam veniam!
            </li>
            <li>
              Asperiores, debitis eius harum, id incidunt natus quisquam quos repellat tempore ut veniam voluptas. Adipisci aliquam deleniti ea iusto quasi totam voluptatum!
            </li>
            <li>
              Ad cupiditate, eos hic obcaecati voluptas voluptatibus! Accusamus eaque laudantium modi quae repellat reprehenderit, vel! Culpa doloremque eligendi excepturi minima nam sed!
            </li>
          </ul>
        </div>
        <div
          class="col-md-4"
        >
          <todo-component />
          <h2>
            Heading
          </h2>
          <p>
            Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui. 
          </p>
          <hr />
          <h2>
            Heading
          </h2>
          <p>
            Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui. 
          </p>
          <hr />
          <h2>
            Heading
          </h2>
          <p>
            Donec sed odio dui. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Vestibulum id ligula porta felis euismod semper. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.
          </p>
        </div>
      </div>
      <hr />
      <footer>
        <p>
          Â© 2016 Company, Inc.
        </p>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
